### ISSUE TITLE
I can't seem to get adjust_text to work with horizontal bar charts

### ISSUE THREAD

**Issue by dcontris** [Phlya/adjustText#29]
```



%matplotlib inline
def add_data_labels(
    ax,
    labels_axis='x',
    x_offset=0.0,
    y_offset=0.0,
    N=1,
    label_value_func=None,
    replace_empty_with='0%',
    *args
    ):
    import numpy as np


    labels=[]
    rects=ax.patches# Assuming is a bar chart. TODO: make this work with scatters, etc.

    #Create tuples of rect and position, with position determined by axis to provide labels for
    for rect, pos in zip(rects, [rect.get_y() for rect in rects] if labels_axis == 'x' else [rect.get_x() for rect in rects]):

        #Get length and height of each rectangle
        length=rect.get_width() if labels_axis == 'x' else rect.get_height()
        height=rect.get_height() if labels_axis == 'x' else rect.get_width()

        if not label_value_func:
            #TODO: Make this function manipulable via another function. i.e. I want to set the function to the default function()*100
            if labels_axis == 'x':
                label_value_func = lambda rect, N: str(int(((rect.get_x() + length) / N) * 100)) + '%'
            else:
                label_value_func = lambda rect, N: str(int(((rect.get_y() + length) / N) * 100)) + '%'

        #Bars with no length should be 0, not nan
        if np.isnan(length):
            length = 0.0
            label_value_func = lambda rect, N: replace_empty_with

        f=label_value_func

        #Add label for each rectangle to a list
        if labels_axis=='x':
            labels.append(ax.text(x = length+x_offset, y = (pos + height / 2.)+y_offset,
                    s = f(rect, N), ha = 'left', va = 'center'))
        else:
            labels.append(ax.text(y = length+y_offset, x = (pos + height / 2.)+x_offset,
                    s = f(rect, N), ha = 'center', va = 'bottom'))

    #TODO: Add support for 'adjustText'
    return ax

import seaborn as sns
from adjustText import adjust_text
sns.set_style("whitegrid")
tips = sns.load_dataset("tips")
ax = sns.barplot(x="tip", y="day", data=tips)
add_data_labels(ax)
adjust_text(ax.texts)

```

The labels seem to have the correct X position in the end, but it moves every text to the very top of the plot.

**Comment by dcontris** [comment]
Sorry... Not sure why the second chunk of code isn't formatted as such.

**Comment by Phlya** [comment]
I edited it to fix the formatting, please check it is all as intended.
I can have a look tomorrow, could you please make sure that your code runs as a complete example? Currently I get an error:
```
NameError                                 Traceback (most recent call last)
<ipython-input-2-5be4e2547885> in <module>()
----> 1 def add_data_labels(ax, labels_axis='x', N=len(data), estimator=lambda x, y: str(int((x/y)*100))+'%'):
      2 
      3     #Labels correspond to X or Y axis values?
      4     if labels_axis == 'x':
      5         labels = lambda x: x.get_x() + length

NameError: name 'data' is not defined
```

**Comment by dcontris** [comment]
Thanks for the prompt reply!
I've fixed the code so it runs as a complete example. Sorry about that!

**Comment by Phlya** [comment]
OK, I figured out the problem - apparently for some reason seaborn inverts the y axis of the plot, and the function gets confused and when it is trying to put all texts inside the axes it actually forces them outside... So simply add `ax.invert_yaxis()` before (and after, if you want the original order of bars) calling adjust_text.

**Comment by Phlya** [comment]
Please let me know whether this fixes your issue, thanks,

**Comment by dcontris** [comment]
Awesome. I'm looking forward to trying this.
I'll give it a shot on Monday

**Comment by dcontris** [comment]
So, your work around seems to work perfectly!
Thanks a lot for figuring this out for me and for this module in general.

You know, it's interesting... When I came to try this today, I actually ran the script to generate the chart that got me onto this whole issue, and without changing anything, it ended up working all by itself.... very weird. I didn't even restart the kernel. With the exact same kernel running that I had last Friday, same variables and everything it just worked...
